apps/client/src/common/hooks/useEventAction.ts

```diff
@@ -111,18 +110,15 @@ export const useEventAction = () => {
       await queryClient.cancelQueries({ queryKey: RUNDOWN });
 
       // Snapshot the previous value
-      const previousData = queryClient.getQueryData<GetRundownCached>(RUNDOWN);
+      const previousData = queryClient.getQueryData<RundownCached>(RUNDOWN);
+      const eventId = newEvent.id;
 
-      if (previousData) {
+      if (previousData && eventId) {
         // optimistically update object
-        const optimisticRundown = [...previousData.rundown];
-        const index = optimisticRundown.findIndex((event) => event.id === newEvent.id);
-        if (index > -1) {
-          // @ts-expect-error -- we expect the event types to match
-          optimisticRundown[index] = { ...optimisticRundown[index], ...newEvent };
-
-          queryClient.setQueryData(RUNDOWN, { rundown: optimisticRundown, revision: -1 });
-        }
+        const newRundown = { ...previousData.rundown };
+        // @ts-expect-error -- we expect the events to be of same type
+        newRundown[eventId] = { ...newRundown[eventId], ...newEvent };
```

Comment:
suggestion: I usually prefer to Type cast instead of expecting a ts error, I still don't love type casting but since we know the types we match maybe we could do

```
        const newRundown = { ...previousData.rundown } as NormalisedRundown;
        newRundown[eventId] = { ...newRundown[eventId], ...newEvent } as OntimeRundownEntry;
```